src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -243,66 +243,72 @@ struct AccessControlEntryCodec
                 break;
             }
             case to_underlying(Fields::kSubjects): {
-                TLV::TLVType subjectsContainer;
-                VerifyOrReturnError(TLV::kTLVType_Array == aReader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
-                ReturnErrorOnFailure(aReader.EnterContainer(subjectsContainer));
-                while ((err = aReader.Next()) == CHIP_NO_ERROR)
+                if (aReader.GetType() != TLV::kTLVType_Null)
```

Comment:
Is there a reason not to do:
```
  DataModel::Nullable<DataModel::List<uint64_t>> subjects;
  ReturnErrorOnFailure(DataModel::Decode(aReader, subjects));
  if (!subjects.IsNull()) {
    // iterate subjects, etc
  }
```
instead of hand-rolling the list decoding?  Bonus points for using the decltype of the generated `subjects` member of AccessControlEntry instead of hardcoding the type, so as to avoid more mismatches...

Reply:
Avoids decoding an intermediate list.

Reply:
Meaning what?  All "decoding a list" means in this case is "positioning a TLV::Reader on the list".  So you save on TLV::Reader instance on the stack.  That's it.